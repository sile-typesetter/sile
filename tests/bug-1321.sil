\begin[direction=RTL,papersize=a7]{document}
\nofolios
\neverindent
\script[src=packages/footnotes]
\font[family=Amiri]
عَرَبي\footnote{عَرَبي}
\begin{script}

SILE.documentState.documentClass:declareFrame("inherit", {
    left = "left(content)",
    right = "right(content)",
    top = "top(content) + 20%ph",
    bottom = "top(content) + 30%ph"
  })

SILE.documentState.documentClass:declareFrame("setleft", {
    left = "left(content)",
    right = "right(content)",
    top = "top(content) + 40%ph",
    bottom = "top(content) + 50%ph",
    direction = "LTR"
  })

SILE.typesetNaturally(SILE.getFrame("folio"), function()
  SILE.typesetter:typeset("عَرَبي")
end)

SILE.typesetNaturally(SILE.getFrame("setleft"), function()
  SILE.typesetter:typeset("foo")
end)

SILE.typesetNaturally(SILE.getFrame("inherit"), function()
  SILE.typesetter:typeset("عَرَبي")
end)

\end{script}
\end{document}
