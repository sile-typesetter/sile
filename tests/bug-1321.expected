Set paper size 	209.7637818	297.6377985
Begin page
Mx 	181.5556
My 	284.1868
Set font 	Amiri;10;400;;normal;;RTL
Mx 	4.7600
Mx 	2.5800
Mx 	5.6100
Mx 	4.7700
T	5147 a=4.7600 5064 a=2.5800 430 x=0.7100 y=-4.6400 3119 a=4.7300 x=0.8800 430 x=-0.7600 2055 a=4.7700	(عَرَبي)
Mx 	10.4882
My 	140.7470
Set font 	Amiri;10;400;;normal;;LTR
Mx 	2.8900
Mx 	4.9700
Mx 	4.9700
T	73 a=3.0000 82 a=4.9700 82 a=4.9700	(foo)
Mx 	181.5556
My 	84.7694
Set font 	Amiri;10;400;;normal;;RTL
Mx 	4.7600
Mx 	2.5800
Mx 	5.6100
Mx 	4.7700
T	5147 a=4.7600 5064 a=2.5800 430 x=0.7100 y=-4.6400 3119 a=4.7300 x=0.8800 430 x=-0.7600 2055 a=4.7700	(عَرَبي)
Mx 	181.5556
My 	25.2419
Mx 	4.7600
Mx 	2.5800
Mx 	5.6100
Mx 	4.7700
T	5147 a=4.7600 5064 a=2.5800 430 x=0.7100 y=-4.6400 3119 a=4.7300 x=0.8800 430 x=-0.7600 2055 a=4.7700	(عَرَبي)
Mx 	178.2040
My 	22.3019
Set font 	Amiri;6.3;400;;normal;;LTR
T	20 w=3.3516	(1)
Mx 	194.4876
My 	263.7880
Set font 	Amiri;9;400;;normal;;LTR
T	20 w=4.7880	(1)
Mx 	192.6696
Set font 	Amiri;9;400;;normal;;RTL
T	17 w=1.8180	(.)
Mx 	158.7216
Mx 	4.2840
Mx 	2.3220
Mx 	5.0490
Mx 	4.2930
T	5147 a=4.2840 5064 a=2.3220 430 x=0.6390 y=-4.1760 3119 a=4.2570 x=0.7920 430 x=-0.6840 2055 a=4.2930	(عَرَبي)
End page
Finish
